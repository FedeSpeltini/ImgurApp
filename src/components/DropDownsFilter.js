import React from 'react';
 import { useFormik } from 'formik';
 
 export const DropDownsFilter = ( { setSection } ) => {

    // const onItemClick = (event) => {
    //     setSection(event.target.value)       
    // }
   const formik = useFormik({
     initialValues: {
        radioSection: 'hot',
        showViral: 'false',
     },
     onSubmit: values => {
       
       setSection(values)
     },
   });


   return (
     <form onSubmit={formik.handleSubmit}>
       <div className="form-check">
            <input className="form-check-input" type="radio"  name="radioSection" onChange={formik.handleChange} id="radioHot" value="hot" defaultChecked></input> 
            <label className="form-check-label" htmlFor="radioHot">
                Hot
            </label>
        </div>
        <div className="form-check">
            <input className="form-check-input" type="radio" name="radioSection" onChange={formik.handleChange} id="radioTop"  value="top"></input>
            <label className="form-check-label" htmlFor="radioTop">
                Top
            </label>
        </div>
        <div className="form-check">
            <input className="form-check-input" type="radio" name="radioSection" onChange={formik.handleChange} id="radioUser"  value="user"></input>
            <label className="form-check-label" htmlFor="radioUser">
                User
            </label>
        </div>

        <div className="form-check">
            <input className="form-check-input" type="checkbox" name="showViral" onChange={formik.handleChange} id="checkViral" value="true"></input>
            <label className="form-check-label" htmlFor="checkViral">
                ShowViral
            </label>
        </div> 
       <button type="submit" className="btn btn-outline-info">Submit</button>
     </form>
   );
 };