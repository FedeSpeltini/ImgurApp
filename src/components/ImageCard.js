import React from 'react';
import { Link } from 'react-router-dom';
// import PropTypes from 'prop-types';
import {PropTypes} from "prop-types";
export const ImageCard = ( {title, urlPresentation, url, boolGif, likes, dislikes, score, description } ) => {
    return (
        <div className="card animate__animated animate__fadeIn" style={ { maxWidth: 250, maxHeight: 500 } }>
            <div className="row no-gutters">
                <div className="col-md-12">
                    { boolGif
                        ? <video src={urlPresentation} type="video/mp4" alt={title} className="card-img" width="250" height="250"  autoPlay controls muted></video>
                        : <img src={urlPresentation} alt={title} className="card-img" width="250" height="250"/>
                    }
                </div>
                <div className="col-md-8">
                    
                    <div className="card-body">
                        <h5 className="card-title"> { title.length > 30 ? title.substring(0,27) + '...' : title } </h5>
                        
                        <Link to={{
                            pathname: `./details` ,
                            state: {
                                urlValue: url, titleValue: title, boolVideo: boolGif, likes: likes, dislikes: dislikes, score: score, description: description
                            }
                            }}>Details
                        </Link>

                    </div>

                </div>
            </div>
        </div>
    )

}

ImageCard.propTypes = {
    title: PropTypes.string.isRequired,
    urlPresentation: PropTypes.string.isRequired,
   // url: PropTypes.json.isRequired,
    boolGif: PropTypes.bool.isRequired,
    likes: PropTypes.number.isRequired,
    dislikes: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired
}
