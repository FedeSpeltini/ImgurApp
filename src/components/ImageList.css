
  .card-columns {
      /* column-count: 4; */
      display: flex; 
      flex-direction: row;
      flex-wrap: wrap; 
  }
  