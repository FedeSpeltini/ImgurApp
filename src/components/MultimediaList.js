import React from 'react'


  export const MultimediaList = ( { type, link, description } ) => {
    return (
        <>
            {
                (type.includes("image") )
                    && <img style={{"margin-bottom": "3%"}} src={link} alt={description} className="card-img"/>        
            }
            {
                (type.includes("video"))
                    && <video style={{"margin-bottom": "3%"}} src={link} alt={description} type={type} className="card-img"  autoPlay controls muted></video>
            }  
            <p>{description}</p>          
        </>
    )

}
