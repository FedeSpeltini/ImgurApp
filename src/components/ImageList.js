import React, { useState } from 'react';

import { useFecthImages } from '../hooks/useFecthImages';
import { DropDownsFilter } from './DropDownsFilter';
import { ImageCard } from './ImageCard';
import  './ImageList.css';
//import { ImageListCss } from './ImageList.css';
export const ImageList = () => {

    const intialState = { radioSection: 'hot', showViral: 'false'}
    const [section, setSection] = useState(intialState);

    const {data:images, loading} = useFecthImages( section );

    return (
        <>

            <button className="btn btn-outline-info" style={{"marginBottom": "5%"}} type="button" data-toggle="collapse" data-target="#collapseExample" aria-expanded="false" aria-controls="collapseExample">
                Show Filters
            </button>
            <div className="collapse" id="collapseExample">
                <DropDownsFilter setSection = { setSection }/>
            </div>
            
            <br></br>
            { loading && <p className="animate__animated animate__flash">Loading...</p> }
            <div className="card-columns d-flex justify-content-center">               
            {
                images.map( image => (
                    <ImageCard 
                            key={ image.id }
                            { ...image }
                        />
                    
                ))
            } 
            </div>

        </>
           
    )
}
