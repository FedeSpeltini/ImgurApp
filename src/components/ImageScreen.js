import React, { Component } from 'react'
import { MultimediaList } from './MultimediaList';


export class ImageScreen extends Component {
    render() {
        const { urlValue, titleValue, boolVideo, likes, dislikes, score, description } = this.props.location.state;
      return (
        <div className="row mt-5">
            <div className="col-4">
                { !Array.isArray(urlValue)
                    ?   [
                            (boolVideo
                                ? <video  src={urlValue} type="video/mp4" alt={titleValue} className="card-img"  autoPlay controls muted></video>
                                : <img src={urlValue} alt={titleValue} className="card-img"/>
                            )
                        ]
                    :   urlValue.map( img => (
                            <MultimediaList 
                                    key={ img.id }
                                    { ...img }
                                />
                        ))
                }
            </div>
            
            <div className="col-8 animate__animated animate__fadeIn">
                <h3 style={{"margin-left": "20%"}}> { titleValue } </h3>
                {/* <h3 className="d-flex justify-content-center"> { titleValue } </h3> */}
                <br></br><br></br>
                
                <h4 style={{"margin-left": "20%"}}> { description } </h4>
                <br></br>

                <div className="d-flex align-items-stretch">   
                    
                    <img src="https://www.flaticon.com/svg/static/icons/svg/889/889221.svg" alt="likes" style={{"margin-left": "20%"}} width="30" height="30"/>
                
                    <h5 style={{"margin-left": "3%"}}>{ likes }</h5> 
                
                    <img src="https://www.flaticon.com/svg/static/icons/svg/889/889220.svg" alt="dislikes" width="30" style={{"margin-left": "5%"}} height="30"/> 
                
                    <h5 style={{"margin-left": "3%"}}>{ dislikes }</h5>

                    <h5 style={{"margin-left": "8%"}}><b>Score</b>: { score }</h5>
                                        
                </div> 
                <br></br><br></br>
                <div className="row">
                    <div className="col-8 pull-right" style={{"margin-left": "20%"}}>
                        {/* <h3 class="one">Text</h3>
                        <button class="btn btn-secondary float-right">Button</button> */}

                        <button 
                            className="btn btn-outline-info  d-flex align-items-end"
                            //onClick={ this.props.history.goBack() }
                            onClick={ () => this.props.history.goBack() }
                        >
                            Return
                        </button>
                    </div>
                </div>

                
            </div>

        </div>
      );
    }
  }
  
 