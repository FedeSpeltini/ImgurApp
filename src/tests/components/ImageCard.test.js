import React from 'react';
import { shallow } from 'enzyme';
import { ImageCard } from "../../components/ImageCard";


describe('should show one picture', () => {

    const title = 'Some Title';
    const urlPresentation   = 'https://localhost/something.jpg';
    const url   = {type: "card-img", link: "https://localhost/something.jpg"};
    const boolGif = false;
    const likes = 10;
    const dislikes = 4;
    const description = "some description";
    const wrapper = shallow( <ImageCard title={ title } urlPresentation={ urlPresentation }  url={ url } boolGif = { boolGif } likes = { likes } dislikes = { dislikes } description = { description }/> );

    test('debe de mostrar el componente correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    })
    
})
