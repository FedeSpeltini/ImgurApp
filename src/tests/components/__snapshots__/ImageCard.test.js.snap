// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` debe de mostrar el componente correctamente 1`] = `
<div
  className="card"
  style={
    Object {
      "maxWidth": 540,
    }
  }
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-7"
    >
      <img
        className="card-img"
        height="250"
        src="https://localhost/something.jpg"
        width="400"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
           
          Some Title
           
        </h5>
        <Link
          to={
            Object {
              "pathname": "./details",
              "state": Object {
                "boolVideo": false,
                "description": "some description",
                "dislikes": 4,
                "likes": 10,
                "titleValue": "Some Title",
                "urlValue": Object {
                  "link": "https://localhost/something.jpg",
                  "type": "card-img",
                },
              },
            }
          }
        >
          Details
        </Link>
      </div>
    </div>
  </div>
</div>
`;

exports[`should show one picture debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
  style={
    Object {
      "maxHeight": 400,
      "maxWidth": 400,
    }
  }
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-7"
    >
      <img
        className="card-img"
        height="250"
        src="https://localhost/something.jpg"
        width="400"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
           
          Some Title
           
        </h5>
        <Link
          to={
            Object {
              "pathname": "./details",
              "state": Object {
                "boolVideo": false,
                "description": "some description",
                "dislikes": 4,
                "likes": 10,
                "score": undefined,
                "titleValue": "Some Title",
                "urlValue": Object {
                  "link": "https://localhost/something.jpg",
                  "type": "card-img",
                },
              },
            }
          }
        >
          Details
        </Link>
      </div>
    </div>
  </div>
</div>
`;
